application/core/MY_Base.php

```diff
@@ -5,10 +5,16 @@ class MY_Base extends CI_Controller
 {
     function __construct()
     {
-        parent::__construct();
-        $db_obj = $this->load->database('default', TRUE);
-        if(!$db_obj->conn_id) {
-            exit('Unable to connect with database with given db details.');
+        try {
+            parent::__construct();
+            $db_obj = $this->load->database('default', TRUE);
+            if(!$db_obj->conn_id) {
+                header($_SERVER['SERVER_PROTOCOL'] . ' 500 Internal Server Error', true, 500);
+                exit('Unable to connect with database with given db details.');
+            }
+        } catch(Exception $e) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 500 Internal Server Error', true, 500);
+            exit('Unable to connect with database with given db details : '.$e->getMessage());
```

Comment:
```suggestion
            exit('Failed to connect with database with given db details : '.$e->getMessage());
```

Reply:
Dont output the exception message in production environment.

Reply:
Done